streams/src/main/java/org/apache/kafka/streams/StoreQueryParameters.java

```diff
@@ -25,8 +25,8 @@
  */
 public class StoreQueryParameters<T> {
 
-    private Integer partition;
-    private boolean staleStores;
+    private final Integer partition;
+    private final boolean staleStores;
```

Comment:
Wondering who this is possible? Should the build not fail for this case? \cc @vvcephei

Reply:
As far as I can see, these fields are only assigned at constructor, and are never modified. Being this, together with in place initialization, the only valid patterns for final fields.
Or am I missing something?

Reply:
The checkstyle rule used only checks local variables, not member fields:
```
    <!-- variables that can be final should be final (suppressed except for Streams) -->
    <module name="FinalLocalVariable">
      <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
      <property name="validateEnhancedForLoopVariable" value="true"/>
    </module>
``` 
See https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.html

Reply:
Yeah, it's a limitation of checkstyle, which makes me a bit sad.